src/components/UserForm.tsx

```diff
@@ -0,0 +1,76 @@
+import { Dispatch, FC, SetStateAction, useState } from "react";
+import Button from "./Button";
+import EmailInput from "./EmailInput";
+import PasswordInput from "./PasswordInput";
+import Select from "./Select";
+
+interface UserFormProps {
+  title: string;
+  email: string;
+  setEmail: Dispatch<SetStateAction<string>>;
+  password: string;
+  setPassword: Dispatch<SetStateAction<string>>;
+  role?: string;
+  setRole?: Dispatch<SetStateAction<string>>;
+  status?: string;
+  setStatus?: Dispatch<SetStateAction<string>>;
+  onClick: () => Promise<void>;
+  buttonText: string;
+  children?: React.ReactNode;
+}
+
+const UserForm: FC<UserFormProps> = ({
```

Comment:
Have you considered using Formik? https://formik.org/docs/overview See examples in my MPP project

Reply:
I use it for destination forms in the other branch. I will refactor this one as well to use it.